src/Groups/perm.jl

```diff
@@ -226,30 +226,35 @@ end
 # takes as input a list of vectors (not necessarily disjoint)
 @doc raw"""
     cperm(L::AbstractVector{<:T}...) where T <: IntegerUnion
+    cperm(L::AbstractVector{<:AbstractVector{T}}) where T <: IntegerUnion
     cperm(G::PermGroup, L::AbstractVector{<:T}...)
-    cperm(L::Vector{Vector{T}}) where T <: IntegerUnion
-    cperm(g::PermGroup,L::Vector{Vector{T}}) where T <: IntegerUnion
+    cperm(G::PermGroup, L::AbstractVector{<:AbstractVector{T}}) where T <: IntegerUnion
 
 For given lists $[a_1, a_2, \ldots, a_n], [b_1, b_2, \ldots , b_m], \ldots$
 of positive integers, return the
 permutation $x = (a_1, a_2, \ldots, a_n) * (b_1, b_2, \ldots, b_m) * \ldots$.
 Arrays of the form `[n, n+1, ..., n+k]` can be replaced by `n:n+k`.
 
-The parent of $x$ is `G`.
-If `G` is not specified then the parent of $x$ is set to
-[`symmetric_group`](@ref)$(n)$,
-where $n$ is the largest integer that occurs in an entry of `L`.
+The parent of $x$ is `G`. If `G` is not specified then the parent of $x$ is
+set to [`symmetric_group`](@ref)$(n)$, where $n$ is the largest integer that
+occurs in an entry of `L`.
+However this incurs non-trivial overhead and so it is generally better
+to provide `G` explicitly.
 
-An exception is thrown if $x$ is not contained in `G`
-or one of the given vectors is empty or contains duplicates.
+An exception is thrown if $x$ is not contained in `G`,
+if one of the given vectors is empty or contains duplicates, or
+if the cycles are not disjoint.
+
+See also [`perm`](@ref) and [`@perm`](@ref) for other ways to create
+permutations.
 
 # Examples
 ```jldoctest
 julia> cperm([1,2,3],4:7)
 (1,2,3)(4,5,6,7)
 
 julia> cperm([1,2],[2,3])
-(1,3,2)
+ERROR: ArgumentError: cycles are not disjoint
```

Comment:
So technically this is breaking. I am actually surprised we documented overlapping cycles as allowed, I thought it was just an implementation artifact. Hrm.

The problem with allowing this is that it kind of defeats the purpose of this function, which is to provide an efficient way to construct permutations from their cycle decomposition. Allowing overlapping cycles means we need to multiple the cycles instead of just writing down the permutation in memory in one go.

Reply:
This situation is similar to that where one creates an element of a pc group from an exponent vector:
If the exponents are in the allowed ranges then no work is needed; the GAP function for that is `PcElementByExponents`. If the exponents may lie outside the allowed ranges then some computations are needed in order to create a valid element; the GAP function for that is `LinearCombinationPcgs`.

I am not sure whether we shall support the creation of permutations from overlapping cycles, but if yes then having two functions would be the solution: a fast one that assumes non-overlapping cycles, and a slow one that may have to do more computations.